lib/ret_web/controllers/api/v1/avatar_controller.ex

```diff
@@ -172,8 +191,20 @@ defmodule RetWeb.Api.V1.AvatarController do
     conn |> send_resp(401, "You do not own this avatar")
   end
 
-  defp delete(conn, %Avatar{} = avatar, %Account{}) do
-    case Repo.delete(avatar) do
+  defp delete(conn, %Avatar{} = avatar, %Account{} = account) do
```

Comment:
I'd suggest moving this into the `avatar_listing.ex` schema and also writing a quick unit test

Reply:
(for the test, important to add additional untouched avatars + listings to ensure no unintentional side effects)

Reply:
Hmm the primary thing being "deleted" here is an avatar, it just happens to also touch listings so I think I will put it in avatar.ex